File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    answer = [0, 0]
    
    
    for a in keyinput:
        if a == "left" and answer[0] > - (board[0]-1)/2 :
            answer[0] = answer[0] - 1
        elif a == "right" and answer[0] < (board[0]-1)/2 :
            answer[0] = answer[0] + 1
        elif a == "up" and answer[1] < (board[1]-1)/2 :
            answer[1] = answer[1] + 1
        elif a == "down" and answer[1] > - (board[1]-1)/2 :
            answer[1] = answer[1] - 1
    

    
    return answer

File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    var answer = [0, 0];
    for (var i of keyinput) {
        if (i == "left" && answer[0] > - (board[0]-1)/2){
            answer[0] -= 1;
        }
        else if (i == "right" && answer[0] < (board[0]-1)/2) {
            answer[0] +=1;
        }
        else if (i == "up" && answer[1] < (board[1] -1 )/2 ) {
            answer[1] += 1;
                 }
        else if (i == "down" && answer[1] > -(board[1]-1)/2 ) {
            answer[1]-=1;
        }
    }
    return answer;
}
